src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala

```diff
@@ -0,0 +1,271 @@
+    var minAreaVaf: Int = 5
+
+    @Args4jOption(name = "--min-occurrence", required = false, usage = "Minimum occurrences to include a kmer ")
+    var minOccurrence: Int = 3
+
+  }
+
+  object Caller extends SparkCommand[Arguments] {
+    override val name = "germline-assembly"
+    override val description = "call germline variants by assembling the surrounding region of reads"
+
+    /**
+     * Extract alignments that are mismatches, insertions or deletions
+     * @param alignment
+     * @return Filtered set of alignments that are mismatches, insertions or deletions
+     */
+    def getVariantAlignments(alignment: ReadAlignment): Seq[(AlignmentState, Int)] = {
+      alignment
+        .alignments
+        .zipWithIndex
+        .filter(tup => AlignmentState.isGapAlignment(tup._1) || tup._1 == AlignmentState.Mismatch)
+    }
+
+    /**
+     *
+     * @param graph An existing DeBruijn graph of the reads
+     * @param currentWindow Window of reads that overlaps the current loci
+     * @param kmerSize Length kmers in the DeBruijn graph
+     * @param minOccurrence Minimum times a kmer must appear to be in the DeBruijn graph
+     * @param expectedPloidy Expected ploidy, or expected number of valid paths through the graph
+     * @return A set of variants in the window
+     */
+    def discoverHaplotypes(graph: Option[DeBruijnGraph],
+                           currentWindow: SlidingWindow[MappedRead],
+                           kmerSize: Int,
+                           reference: ReferenceGenome,
+                           minOccurrence: Int = 3,
+                           expectedPloidy: Int = 2): (Option[DeBruijnGraph], Iterator[CalledAllele]) = {
+
+      val locus = currentWindow.currentLocus
+      val reads = currentWindow.currentRegions()
+
+      // TODO: Should update graph instead of rebuilding it
+      // Need to keep track of reads removed from last update and reads added
+      val currentGraph: DeBruijnGraph = DeBruijnGraph(
+        reads.map(_.sequence),
+        kmerSize,
+        minOccurrence,
+        mergeNodes = true
+      )
+
+      val referenceStart = (reads.head.start).toInt
+      val referenceEnd = (reads.last.end).toInt
+
+      val currentReference: Array[Byte] = reference.getReferenceSequence(
+        currentWindow.referenceName,
+        referenceStart,
+        referenceEnd
+      )
+
+      val referenceKmerSource = currentReference.take(kmerSize)
+      val referenceKmerSink = currentReference.takeRight(kmerSize)
+
+      // If the current window size doesn't cover the kmer size we won't be able to find the reference start/end
+      if (referenceKmerSource.length != kmerSize || referenceKmerSink.length != kmerSize)
+        return (graph, Iterator.empty)
+
+      val paths = currentGraph.depthFirstSearch(
+        referenceKmerSource,
+        referenceKmerSink
+      )
+
+      // Take `ploidy` paths
+      val topPaths = paths
+        .take(expectedPloidy)
```

Comment:
Yea, not even necessarily the two longest but just any two really. Need to come up with some sort of weighting/scoring strategy. I think the better solution will be to take some larger number (6-8) and align them and takes the ones with the best alignments.